node/modules/lp2p/host.go

```diff
@@ -71,16 +70,18 @@ func MockHost(mn mocknet.Mocknet, id peer.ID, ps peerstore.Peerstore) (RawHost,
 	return mn.AddPeerWithPeerstore(id, ps)
 }
 
-func DHTRouting(client bool) interface{} {
+func DHTRouting(mode dht.ModeOpt) interface{} {
 	return func(mctx helpers.MetricsCtx, lc fx.Lifecycle, host RawHost, dstore dtypes.MetadataDS, validator record.Validator) (BaseIpfsRouting, error) {
 		ctx := helpers.LifecycleCtx(mctx, lc)
 
 		d, err := dht.New(
 			ctx, host,
-			dhtopts.Client(client),
-			dhtopts.Datastore(dstore),
-			dhtopts.Validator(validator),
-			dhtopts.Protocols("/lotus/kad/1.0.0"),
+			dht.Mode(mode),
+			dht.Datastore(dstore),
+			dht.Validator(validator),
+			dht.ProtocolPrefix("/lotus"),
+			dht.QueryFilter(dht.PublicQueryFilter),
+			dht.RoutingTableFilter(dht.PublicRoutingTableFilter),
```

Comment:
These filters help ensure that the DHT only queries peers that appear to be on the public internet (instead of behind a NAT or some other private network) and only adds those publicly accessible peers to it's routing table.

Reply:
And they'll likely break most tests because they filter out localhost.